pkg/scheduler/nodeinfo/snapshot/snapshot.go

```diff
@@ -31,13 +35,40 @@ type Snapshot struct {
 	Generation   int64
 }
 
-// NewSnapshot initializes a Snapshot struct and returns it.
-func NewSnapshot() *Snapshot {
+var _ schedulerlisters.SharedLister = &Snapshot{}
+
+// NewEmptySnapshot initializes a Snapshot struct and returns it.
+func NewEmptySnapshot() *Snapshot {
 	return &Snapshot{
 		NodeInfoMap: make(map[string]*schedulernodeinfo.NodeInfo),
 	}
 }
 
+// NewSnapshot initializes a Snapshot struct and returns it.
+func NewSnapshot(pods []*v1.Pod, nodes []*v1.Node) *Snapshot {
+	nodeInfoMap := schedulernodeinfo.CreateNodeNameToInfoMap(pods, nodes)
+	var nodeInfoList []*schedulernodeinfo.NodeInfo
```

Comment:
nit: pre-malloc memory for the list.

Reply:
done.

Reply:
as to the naming, the idea is that plugins will get the sharedlister from the handle and will do something like this:
```
podLister := handle.SnapshotSharedLister().Pods()
```

and so adding Lister suffix like PodLister and NodeInfoLister becomes redundant.

Reply:
yes, it is a little redundant.